brut.apktool/apktool-lib/src/main/java/brut/androlib/src/SmaliDecoder.java

```diff
@@ -17,48 +17,53 @@
 package brut.androlib.src;
 
 import brut.androlib.exceptions.AndrolibException;
-import com.android.tools.smali.baksmali.Baksmali;
-import com.android.tools.smali.baksmali.BaksmaliOptions;
+import com.android.tools.smali.baksmali.Adaptors.ClassDefinition;
+import com.android.tools.smali.baksmali.formatter.BaksmaliWriter;
 import com.android.tools.smali.dexlib2.DexFileFactory;
 import com.android.tools.smali.dexlib2.Opcodes;
 import com.android.tools.smali.dexlib2.dexbacked.DexBackedDexFile;
 import com.android.tools.smali.dexlib2.dexbacked.DexBackedOdexFile;
 import com.android.tools.smali.dexlib2.analysis.InlineMethodResolver;
+import com.android.tools.smali.dexlib2.iface.ClassDef;
 import com.android.tools.smali.dexlib2.iface.DexFile;
 import com.android.tools.smali.dexlib2.iface.MultiDexContainer;
+import com.android.tools.smali.util.ClassFileNameHandler;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Ordering;
 
 import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.logging.Logger;
+
+import javax.annotation.Nullable;
+
+import brut.androlib.Config;
 
 public class SmaliDecoder {
 
-    public static DexFile decode(File apkFile, File outDir, String dexName, boolean bakDeb, int apiLevel)
+    Config config;
+    private final static Logger LOGGER = Logger.getLogger(SmaliDecoder.class.getName());
+
+    public static DexFile decode(File apkFile, File outDir, String dexName, Config mConfig)
             throws AndrolibException {
-        return new SmaliDecoder(apkFile, outDir, dexName, bakDeb, apiLevel).decode();
+        return new SmaliDecoder(apkFile, outDir, dexName, mConfig).decode();
     }
 
-    private SmaliDecoder(File apkFile, File outDir, String dexName, boolean bakDeb, int apiLevel) {
-        mApkFile = apkFile;
-        mOutDir = outDir;
-        mDexFile = dexName;
-        mBakDeb = bakDeb;
-        mApiLevel = apiLevel;
+    private SmaliDecoder(File mApkFile, File mOutDir, String mDexName, Config mConfig) {
+        config = mConfig;
+        config.dexName = mDexName;
+        config.apkFile = mApkFile;
+        config.outDir = mOutDir;
     }
 
     private DexFile decode() throws AndrolibException {
         try {
-            final BaksmaliOptions options = new BaksmaliOptions();
-
             // options
-            options.deodex = false;
-            options.implicitReferences = false;
-            options.parameterRegisters = true;
-            options.localsDirective = true;
-            options.sequentialLabels = true;
-            options.debugInfo = mBakDeb;
```

Comment:
most of these variables are already present in BaksmaliOptions, and are now also configurable with command-line-flags, so no need to hardcode anymore :)